swap_meet/vendor.py

```diff
@@ -0,0 +1,201 @@
+            item (str): The item to be added. 
+
+        Returns:
+            item (str): The item that was added.
+        """
+
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        """Removes the matching `item` from the `inventory`.
+
+        Args:
+            item (str): The item that will be removed.
+
+        Returns:
+            item (str): The item that was removed. If there is no matching item
+                in the `inventory`, returns `False`.
+        """
+
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+        """Iterates over the `inventory` to find the categories of the items in that list. 
+
+        Args:
+            category (str): Represents the `category` of the items.
+
+        Returns:
+            same_category_items: A list of `Item`s in the `inventory` with that
+                `category`.
+        """
+
+        same_category_items = []
+        for item in self.inventory:
+            if item.category == category:
+                same_category_items.append(item)
+        return same_category_items
+
+    def swap_items(self, friend, my_item, their_item):
+        """Removes `my_item` from this `Vendor`'s `inventory` and adds it to the `friend`'s `inventory`.
+
+        Args:
+            friend (object): An instance of another `Vendor`, representing the
+                friend that this `Vendor` instance is swapping with.
+
+            my_item (str): Represents the item this `Vendor` instance plans to
+                give.
+
+            their_item (str): Represents the item the `friend` plans to give.
+
+        Returns:
+            bool: If this `Vendor`'s `inventory` doesn't contain `my_item` or
+                the `friend`'s `inventory` doesn't contain `their_item`,
+                returns `False`; otherwise, returns `True`.
+        """
+
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.remove(my_item)
+            friend.add(my_item)
+            friend.remove(their_item)
+            self.add(their_item)
+            return True
+        return False
+
+    def swap_first_item(self, friend):
+        """Swaps the first item in this `Vendor`'s `inventory` with the first item in the `friend`'s `inventory`.
+
+        Args:
+            friend (object): An instance of another `Vendor`, representing the
+                friend that this `Vendor` instance is swapping with.
+
+        Returns:
+            bool: If either itself or the `friend` have an empty `inventory`,
+                returns `False`. Otherwise, returns True.
+        """
+
+        if not self.inventory or not friend.inventory:
+            return False
+        else:
+            my_item = self.inventory.pop(0)
+            self.add(friend.inventory.pop(0))
+            friend.add(my_item)
+            return True
```

Comment:
This looks a lot like the code in the method above. It may be worth seeing if you can revisit this to DRY up your code by calling the other method from this one

Reply:
So it took me a while to understand how I could be able to use the swap here, in the beginning it seemed that it wouldn't do exactly what I wanted until I figured out (with the help of my tutor) that I should just index the first item in each vendors' lists on method call :)) will change this!